lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -3130,13 +3064,39 @@ void VisitNestedScopes(ParseNode* pnodeScopeList, ParseNode* pnodeParent, ByteCo
                 i = 0;
                 ParseNodePtr containerScope = pnodeScope->sxFnc.pnodeScopes;
 
-                VisitNestedScopes(containerScope, pnodeScope, byteCodeGenerator, prefix, postfix, &i);
+                if (pnodeScope->sxFnc.HasNonSimpleParameterList() && !pnodeScope->sxFnc.funcInfo->GetParamScope()->GetCanMergeWithBodyScope())
+                {
+                    // Pop the body scope
+                    byteCodeGenerator->PopScope();
```

Comment:
Where was the body scope pushed prior to this point? Is this situation similar to the one that we fixed in the Emit pass?